baybe/acquisition/base.py

```diff
@@ -84,6 +86,15 @@ def to_botorch(
             additional_params["mc_points"] = to_tensor(
                 self.get_integration_points(searchspace)  # type: ignore[attr-defined]
             )
+        if pending_experiments is not None:
+            if self.is_mc:
+                pending_x = searchspace.transform(pending_experiments, allow_extra=True)
```

Comment:
The `allow_extra` part should be removed here

Reply:
its allowed for measurements, why not for pending measurements?

Reply:
Hm, perhaps we need to talk about what we want to achieve here. First, we need to distinguish between "extra" in the sense of "extra in context of the object" and "extra in general". The reason why I put "allow_extra" for the measurements is because otherwise I could simply not pass the `measurements` dataframe (which also contains target columns) to `searchspace.transform`. However, for the pending measurements, this is possible since no extra target columns present.

Now, the question is rather: How do we want to handle a situation where `measurements` contains completely unrelated columns? To me, this still strongly smells like an unintended case leading to silent bugs, e.g. a user might have added a task-column but forgot to add a corresponding `TaskParameter` -> worst case scenario. In my eyes, we should absolutely handle this case somewhere to avoid the silent bugs (again mentioning my `FixedParameter` idea or similar). If that is cleanly handled, then also the `allow_extra` flag should vanish for the `measurements` and we should rather explicitly downfilter

Reply:
additional columns in the form of metadata seem entirely fair to me, a convenience likely welcomed by users
This level of babysitting you suggest imo doesnt make sense

Reply:
Well, the questions is what is more severe:
* Forcing the user to be explicit by requiring something like `campaign.add_measurements(df.filter(campaign.columns))` (<-- we could easily a convenience property like that) in cases where they need to carry metadata columns
* Getting a completely wrong optimization result that is difficult/impossible to notice, i.e. worst possible scenario with baybe in my opinion? <-- in fact already happened to me. Can easily occur when you are still playing around and adding dimensions/parameters here and there.

Let's also hear @AVHopp's opinion on that

Reply:
even if we agree on a proposed solution, there is no need to hastingly implement this here which would cause and inconsistent treatment

defacto now users can include arb metacolumns in measurmeents --> it is just consistent to also allow that for pending measurements

If that should change, it should change in the same PR

Reply:
For the sake of this PR, I would also say that "If users can do that with normal measurements, they should be able to do that with pending ones", since this is also what I would expect as a user. I would be very confused if I were allowed to meta data columns when adding measurements, but not when adding pending points. In particular, this might even be very relevant here since stuff like "Estimated time to result" might be a column people would *exclusively* add to pending points.

Regarding the general point: Good question if we should enable this in general/design it in such a way that silent bugs are avoided. Not sure what would be the best way to handle this, but I think we all agree that this in not within the scope of this PR.

Reply:
I would tentatively consider this then as decide by majority, but ill leave this open for 1 more day in case adrian feels strongly about it

I think its an entirely valid process if you were to open an issue so we remember to address this in a consistent PR in the future

Reply:
No, it's fine, I'm definitely with you that this should be its own PR – if at all. We just happened to stumble over it in the current one. Happy to open an issue so we can close here 👍🏼 

Just for clarification, @AVHopp: I'm not suggesting to have one variable dealt with in one way and the other in a different way. If we decide to include validation, this should then affect all inputs equally 🙃

Reply:
#366

Reply:
resolving here